### PR TITLE
Add limitation functionality

### PR DESCRIPTION
**Pull request by ryanmitchell**
This is the cart related changes for: https://github.com/tastyigniter/TastyIgniter/issues/481

### CODE REVIEW

File: cartconditions/Coupon.php [hunk 1]
```diff
@@ -2,6 +2,7 @@
```
Comment (sampoyigi) [inline cartconditions/Coupon.php:2]: This is no longer needed right?

File: classes/OrderManager.php [hunk 2]
```diff
@@ -190,10 +190,8 @@ public function saveOrder($order, array $data)
 
         $order->addOrderMenus(Cart::content()->all());
         $order->addOrderTotals($this->getCartTotals());
-
-        // Lets log the coupon so we can redeem it later
-        if ($couponCondition = Cart::conditions()->get('coupon'))
-            $order->logCouponHistory($couponCondition, $this->customer);
+        
+        Event::fire('igniter.checkout.afterSaveOrder', [$order]);
```
Comment (sampoyigi) [inline classes/OrderManager.php:194]: this here should be enough

**Review by sampoyigi — CHANGES_REQUESTED** [review]
revert the changes

File: cartconditions/Coupon.php (hunk 1 above)
Comment (ryanmitchell) [reply cartconditions/Coupon.php:2]: Yep its not needed - sorry I havent updated this PR as I was waiting to see how we progressed with the flame side of the calculation

File: classes/OrderManager.php (hunk 2 above)
Comment (ryanmitchell) [reply classes/OrderManager.php:194]: Yep

File: classes/OrderManager.php [hunk 3]
```diff
@@ -190,10 +190,8 @@ public function saveOrder($order, array $data)
 
         $order->addOrderMenus(Cart::content()->all());
         $order->addOrderTotals($this->getCartTotals());
-
-        // Lets log the coupon so we can redeem it later
-        if ($couponCondition = Cart::conditions()->get('coupon'))
-            $order->logCouponHistory($couponCondition, $this->customer);
+        
+        Event::fire('igniter.checkout.afterSaveOrder', [$order, $this->customer]);
```
Comment (sampoyigi) [inline classes/OrderManager.php:194]: customer should be available on $order->customer

Comment (ryanmitchell) [reply classes/OrderManager.php:194]: Yeah I had thought that. Was just copying what was there before but makes sense to reduce what is passed.

**Review by sampoyigi — APPROVED** [review]

**Pull request merged**